### ISSUE TITLE
Pandas Fragmentation Warning

### ISSUE THREAD

**Issue by Possums** [mljar/mljar-supervised#505]
Getting this warning when training my model:

```
DataFrame is highly fragmented. This is usually the result of calling `frame.insert` many times, which has poor performance.   Consider joining all columns at once using pd.concat(axis=1) instead.  To get a de-fragmented frame, use `newframe = frame.copy()`
```

Is this something that could be causing performance issues? Using Compete mode, with LightGBM/Xgboost/CatBoost models for a regression task.

**Comment by pplonski** [comment]
@Possums thank you for reporting! Do you have a lot of categorical columns in your data? Do you get results in reasonable time? Do you get good results?

Is your data/code public so I can reproduce the problem? Or maybe you can provide some dummy data to reproduce the issue?

**Comment by Possums** [comment]
Thanks for looking into this!

It's the Numer.ai dataset, so there are a little over 1,000 columns. Each feature takes the values of 0, 0.25, 0.5, 0.75 or 1, but I don't think it's being treated as categorical. Training takes a while (about 2 days, 64 cores) but the results are good.

**Comment by williamty** [comment]
I also met the same error (sorry, the data cannot be provided):
/home/fuwai/.local/lib/python3.9/site-packages/supervised/preprocessing/label_binarizer.py:30: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`

**Comment by pplonski** [comment]
Thanks @williamty for reporting! Anyone interested in PR?
